src/App.tsx

```diff
@@ -1,27 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
-import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const PAGES_NUM = 42;
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [page, setPage] = useState(1);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {page} (items {(page - 1) * perPage + 1} -{' '}
+        {page * perPage < PAGES_NUM ? page * perPage : PAGES_NUM} of {PAGES_NUM}
+        )
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            onChange={(event: React.ChangeEvent<HTMLSelectElement>) => {
+              setPerPage(+event.target.value);
+              setPage(1);
+            }}
+          >
             <option value="3">3</option>
-            <option value="5">5</option>
+            <option value="5" selected={true}>
+              5
+            </option>
```

Comment:
The 'selected' attribute on the <option> element is not the recommended way to set a default value in React. Instead, you should control the selected value with the state of the <select> element by setting its 'value' property to match the corresponding state.